src/main/java/org/embl/mobie/io/ome/zarr/loaders/N5OMEZarrImageLoader.java

```diff
@@ -343,17 +334,32 @@ private double[] getXYZScale( OmeZarrMultiscales.CoordinateTransformations coord
         double[] xyzScale = null;
         if (scale != null && zarrAxesList != null) {
             int scalesFirstIndexBackward = scale.length - 1;
-            if (zarrAxes.containsXYZCoordinates()) {
-                xyzScale = new double[]{
-                        scale[scalesFirstIndexBackward],
-                        scale[scalesFirstIndexBackward - 1],
-                        scale[scalesFirstIndexBackward - 2] };
+            if (zarrAxes.hasChannels()) {
+                if (zarrAxes.is3DWithChannels() || zarrAxes.is2D()) {
```

Comment:
What does `is2D()` return?
I think this should include the case `is3DWithChannels()`, doesn't it?

Reply:
I also find this confusing. I would assume that `is3DWithChannels` applies to the cases where we have axes `CZYX` or `TCZYX`. And in that case the scale that you set does not make sense (because we should also set a scale value for `Z`).

Reply:
fixed